File: Server/farm-manager-api/src/main/java/com/smartfarm/api/entity/GrowthProcess.java
package com.smartfarm.api.entity;

import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "Growth_Processes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GrowthProcess {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "process_id")
    private Long processId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "crop_id")
    private Crop crop;

    @Column(name = "process_name", nullable = false, length = 100)
    private String processName;

    @Column(name = "total_days", nullable = false)
    private Integer totalDays;

    @OneToMany(mappedBy = "process", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Stage> stages;
}

File: Client/my-farm-frontend/src/models/GrowthProcess.ts
import type { StageBase } from "./Stage"

export interface GrowthProcessBase {
    processId: number
    cropId: number
    cropName: string
    variety: string
    processName: string
    totalDays: number
    stages: StageBase[]
}

export interface ProcessDetail {
    processId: number;
    processName: string;
    totalDays: number;
    cropId: number;
    cropName: string;
    variety: string;
    stages: StageBase[];
}